spec/examples/weekly_rule_spec.rb

```diff
@@ -338,6 +338,19 @@ module IceCube
           end
         end
       end
+
+      context 'when base interval validation type is not hourly' do
```

Comment:
Updated language somewhat. The more I dig into this, the more niche this bug seems.

 - schedule with both weekly intervals and hour of day rules to trigger the `base_interval_validation.type` mismatch
 - a `start_time` that is misaligned with the `day` rule so that [start_time modification](https://github.com/seejohnrun/ice_cube/blob/master/lib/ice_cube/schedule.rb#L433-L434) due to `spans: true` is not defaulted back to `start_time`

I can perhaps nest more contexts if that would be preferred?

It's also a bit tricky as to where exactly this spec should exist. The change was made in `hour_of_day`, so I'm happy to relocate it there if that feels more relevant.

The discovery process was very much outside-in, so apologies for any extra confusion.